contrib-03-AnalysisWithPython/Readme.md

```diff
@@ -0,0 +1,153 @@
+```python
+   import numpy as np
+   custom_binning = np.linspace(5000,5500,100) #min, max and number of bins 
+   mass_B0 = df['B0_M'].plot.hist(bins=custom_binning, label = 'mass of the
+   B^{0}_{M} [MeV/c$^{2}$]'} #label of the plot can be written in LaTeX
+```
+
+Then, you just need to add `plt.show()` in your code to plot the histogram.
+A more detailed code is written in both `compare_mc_data.py` and
+`AnalysisWithPython.ipynb`.  
+
+
+## Cutting
+
+One needs to cut either to reduce a dataset or to produce some plot and there is
+a way to both :
+- create a reduced dataset : 
+    ```python
+    cut_string = " B0_M >5250 & B0_P > 500 "
+    df_reduced = df.query(cut_string)
+    ```   
+- select events without creating a new dataset :
+    ```python
+    mask = (df['B0_M'] > 5250) & (df['B0_P'] > 500)
+    df_reduced = df[mask]
+    ```
+Which method to choose will depend on the use-case. And then, the plotting part
+is taken care by :  
+    ```python
+        df.query(cut_string)['B0_M'].plot.hist(bins=custom_binning)
+    ```
+
+## MVA
+
+ROOT users are used to TMVA when they need to train a BDT for their analysis.
+The *scikit-learn* project is the most mature way to such task in Python.
+Posts from Tim Head's
+[blog](http://betatim.github.io/posts/sklearn-for-TMVA-users/) help us a lot to set up a BDT using this library.
+
+```python
+    #For scikit-learn, one needs to concatenate all the data in one dataframe
+    X_test_data =  df[:int(0.2*n_events)]#First 20% of dataset for testing BDT
+    X_test_mc   =  df_mc[:int(0.2*n_events)]
+    X_test      =  df.append(X_test_mc,ignore_index = True).values
+
+    X_train_data =  df[int(0.2*n_events):n_events]#80% of dataset for training
+    BDT
+    X_train_mc   =  df_mc[int(0.2*n_events):n_events]
+    X_train      =  X_train_data.append(X_train_mc,ignore_index = True).values
+
+    # y_test and y_train are used to store categories (1 == signal, 0 ==
+    background)
+    y_test  = int(0.2*n_events)*[1]+int(0.2*n_events)*[0]
+    y_train = int(0.8*n_events)*[1]+int(0.8*n_events)*[0]
+
+    #DecisionTree settings
+    dt = DecisionTreeClassifier(max_depth=3,
+                                min_samples_leaf=0.05*len(X_train))
+
+    #BoostedDecisionTree settings
+    bdt = AdaBoostClassifier(dt, algorithm='SAMME',
+                            n_estimators=800, learning_rate=0.5)
+    #Training
+    bdt.fit(X_train, y_train)
+    
+    #BDT output
+    sk_y_predicted = bdt.predict(X_test) 
+```
+## Adding variable to the tree and saving output
+
+Some analysis step could create a new variable, such as a BDT output, which is to be
+added to a ROOT file. Two steps are needed to perform this :
+- The variable should be added to the dataframe using `df['BDT'] = sk_y_predicted`
+- The modified dataframe can be stored in a new ROOT file using
+  `df.root_pandas.to_root(df, 'output.root', key = 'mytree')`
+
+## Conclusion
```

Comment:
Before the conclusion, I think it would be nice to have a "Going further" (short) section in which you outline how to continue with the analysis, maybe giving links to useful tools for python-based analysis.